### DISCUSSION TITLE
Xarray @ SciPy 2023

### DISCUSSION THREAD

**Discussion by dcherian** [pydata/xarray#7480] [General]
## Timeline

**Proposal Deadline** Mar 01, 2023

**Tutorials** July 10, 11 (Mon, Tues)
**Conference** July 12, 13, 14 (Wed-Fri)
**Sprints** July 15, 16 (Sat, Sun)

## Tutorial

I think we should submit a Xarray tutorial proposal. **Who's interested in organizing/participating/helping?** I emphasize that Xarray community members are _very very welcome_ to help organize, present, and help out with the tutorial in-person! Please sign up in a reply

**One question: Do we want to try an intermediate-level tutorial instead?** This would be good motivation to clean up our intermediate level tutorial material (https://tutorial.xarray.dev/). 

Topics could include
- Computational patterns: Rolling/coarsen/groupby/weighted
- Reading/Combining data: concat/merge/open_mfdataset
- Accessing cloud datasets?
- indexing: vectorized, and orthogonal indexing
- duck array support: wrapping pint, dask (maybe bit more on dask)
- Flexible indexing demo?
- apply_ufunc :scream_cat:

**Comment by andersy005** [comment]
@dcherian, count me in

**Comment by negin513** [comment]
@dcherian Yes, please count me in!

**Comment by TomNicholas** [comment]
I'm interested too :slightly_smiling_face:

**Comment by dcherian** [comment]
Please like this response with a 

-  🎉 for fundamental tutorial
- 🚀 for a intermediate-level tutorial

(i'm voting for both to kick things off)

**Comment by dcherian** [comment]
Here's a proposal [draft](https://docs.google.com/document/d/1BEkPx9dNMG9zRBddeMR_kMLB8ECbuMlYsi46BnjWQd8/edit#heading=h.e6qc3wx71eyw). Please only look at the outline section and add comments. I'll update the rest soon.

If you're interested in developing material, please add your name, affiliation to the author list. **Everyone is welcome to help out!** New and less-experienced contributors are very welcome; the material will be a lot better if its written and/or reviewed by you!

For topics I chose
1. Indexing (advanced, vectorized, boolean) (30 minutes)
2. Computational Patterns (rolling; coarsen; groupby) (50 minutes)
3. Duck Arrays  (50 minutes)
4. apply_ufunc: 30 minutes teachnig + 30 minutes exploration / Q&A(!)

Curious to hear if you think the "Computational patterns"([existing material](https://tutorial.xarray.dev/intermediate/01-high-level-computation-patterns.html)) section should be swapped out for an hour on concat/merge/combine.

**Reply by dcherian** [reply]
That draft is now ready. All comments are welcome.

Again new participants are also welcome.

**Comment by dcherian** [comment]
Yay! this proposal was accepted!

**Reply by TomNicholas** [reply]
Yay! Thank you so much @dcherian !

**Comment by dcherian** [comment]
Here's an [outline](https://github.com/xarray-contrib/xarray-tutorial/issues/170) of the work needed to make this happen. Please comment there if you're interested in helping out.
